src/v/platform/stop_signal.h

```diff
@@ -27,10 +27,6 @@ class stop_signal {
     ~stop_signal() {
         ss::engine().handle_signal(SIGINT, [] {});
         ss::engine().handle_signal(SIGTERM, [] {});
-
-        // We should signal for unit tests, because some background tasks does
-        // not finish without it
-        signaled();
```

Comment:
1. the commit message

    fixture: do not require stop signal to be signalled in destructor

doesn't provide any information about why this change is being made in support of the fixture tests.

2. looking closer--and despite the comment here about signaling for unit tests--isn't this raii behavior relied upon by redpanda application? for example, if an exception occurs during startup will the destruction of the app_signal object signal aborts for sub-systems so that the deferred services stopping will complete?